include/mbedtls/ssl.h

```diff
@@ -42,7 +42,7 @@
 #include "mbedtls/dhm.h"
 #endif
 
-#if defined(MBEDTLS_ECDH_C)
+#if defined(MBEDTLS_ECDH_C) || defined(PSA_WANT_ALG_ECDH)
```

Comment:
As with the guards around `ecdh_ctx` below, `mbedtls/ecdh.h` doesn't declare anything that `PSA_WANT_ALG_ECDH` should need.

Reply:
@gilles-peskine-arm for ssl.h and ssl_internal.h if MBEDTLS_ECDH_C is not defined but PSA_WANT_ALG_ECDH is defined with accelerators the build will not compile if MBEDTLS_ECDSA_C is also defined. What is the appropriate way to resolve that dependency. Should ssl_cli.c and ssl_srv.c be updated to make the use of ecdh_ctx only when ECDH_C is set and not also allow ECDSA_C in those files?